### PR TITLE
Respond with ParamOutgoingResetRequest

### PR DESCRIPTION
**Pull request by algesten**
This is a fix for the issue reported in https://github.com/algesten/str0m/issues/286

### CODE REVIEW

**Comment by algesten** [comment]
This is marked as Draft, since I think we need to check spec here. It might be the close calls shouldn't be automatic in sctp-proto, but a stream from outside `stream.stop()` – looking at pion that _might_ be what we want to do.

@k0nserv is good with the spec detective work.

File: src/association/mod.rs [hunk 1]
```diff
@@ -1856,7 +1856,11 @@ impl Association {
             );
             for id in &p.stream_identifiers {
                 if self.streams.contains_key(id) {
-                    self.unregister_stream(*id);
+                    if respond {
+                        self.send_reset_request(*id).unwrap();
```
Comment (algesten) [inline src/association/mod.rs:1860]: Unwrap here is probably not a good idea.

**Comment by k0nserv** [comment]
The spec seems pretty clear: respond with an OutgoingResetRequest when one is received to confirm the closure. However, I'm not sure Pion implemented it like this in the end, there's a lot of relevant discussion in https://github.com/pion/sctp/pull/238

**Comment by Marcel-G** [comment]
I tried to replicate pion's fix in [webrtc-rs/webrtc](https://github.com/webrtc-rs/webrtc/pull/480) which has the same problem

**Comment by algesten** [comment]
So maybe this solve is more correct. If only we could make it close straight away and not wait for the next RTO.

**Comment by algesten** [comment]
I think this might be ready. The problem was that the response `ParamOutgoingResetRequest` needs to have the `reconfig_request_sequence_number` set. And now it does. The channels close promptly.

**Comment by algesten** [comment]
Hi @rainliu what do you think about this fix?

**Review by rainliu — APPROVED** [review]

**Pull request merged**